pyhorn/client.py

```diff
@@ -35,19 +36,20 @@ def newfunc(*args, **kwargs):
                         "Perhaps it is not available on this node?")
                 elif resp.status_code == 401:
                     raise MHClientHTTPError("Access denied: %s" % e)
+                elif resp.status_code in callbacks:
+                    callbacks[resp.status_code](e)
                 else:
-                    for code, handler in callbacks.items():
-                        if resp.status_code == code:
-                            handler(e)
+                    raise
         return newfunc
     return wrapper
 
 class MHClient(object):
 
-    def __init__(self, base_url, user, passwd):
+    def __init__(self, base_url, user, passwd, timeout=None):
```

Comment:
Here it could be `, timeout=_default_timeout):` too but I tend to prefer cleaner, consistent function definitions + explicitly setting defaults in the body.